### PR TITLE
Wayfarer negative trait addition

### PR DESCRIPTION
**Pull request by Xanderga**
#### Summary
```SUMMARY: Content "Adds new negative trait Wayfarer"```

#### Purpose of change
To add a layer of difficulty for the players who choose to do so as well as changing the dynamic of gameplay by disabling vehicles, and remote controlled vehicles as modes of travel.

#### Describe the solution
Adds a new trait that's used as a boolean to disable the use of control_vehicle(). Similar checks will occur to disable the usage of remote controlled vehicles.

#### Additional context
8 points is figured since the inability to operate vehicles is more debilitating than being unable to read books from the 6 point illiterate trait. Being unable to use vehicles forces the player to either stay in one location (or leave that location only bringing with you what you can carry or drag in a cart), or be constantly on the move to stay alive. It severely limits mobility in all parts of the game, reduces player's escape cards since they can no longer just drive away, and adds a layer of complexity for the players who choose this trait.

edit: Included remote controlled vehicles. No longer a WIP, ready for merge.
Re-PR'd under #34601

### CODE REVIEW

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/is-there-a-mod-to-play-without-a-car/21625/5

**Comment by Qrox** [comment]
Can the player still use remote controls to drive inside the driven vehicle? From my understanding of the code remote control is handled elsewhere in the code, so you might want to disable that as well.

**Comment by Little-119** [comment]
Does this apply to mounting horses and such?

File: data/json/mutations/mutations.json [hunk 1]
```diff
@@ -6394,5 +6394,14 @@
     "description": "Spending any amount of time in familiar places makes you miserable.  Must.  Keep.  Moving.",
     "prereqs": [ "NOMAD2" ],
     "category": [ "CHIMERA" ]
+  },
+  {
+    "type": "mutation",
+    "id": "WAYFARER",
+    "name": "Wayfarer",
+    "points": -8,
+    "description": "Whether from personal choice or childhood trauma, vehicle manipulation is off-limits to you, even if your life depended on it. Walking is your only mode of travel.",
```
Comment (AMurkin) [inline data/json/mutations/mutations.json]:
```suggestion
    "description": "Whether from personal choice or childhood trauma, vehicle manipulation is off-limits to you, even if your life depended on it.  Walking is your only mode of travel.",
```

File: src/game.cpp [hunk 2]
```diff
@@ -9589,7 +9589,9 @@ point game::place_player( const tripoint &dest_loc )
 
     if( vp1.part_with_feature( "CONTROLS", true ) && u.in_vehicle && !u.is_mounted() ) {
         add_msg( _( "There are vehicle controls here." ) );
-        add_msg( m_info, _( "%s to drive." ), press_x( ACTION_CONTROL_VEHICLE ) );
+        if ( !u.has_trait(trait_id( "WAYFARER" ) ) ) {
```
Comment (AMurkin) [inline src/game.cpp]:
```suggestion
        if ( !u.has_trait( trait_id( "WAYFARER" ) ) ) {
```

File: src/game.cpp [hunk 3]
```diff
@@ -9589,7 +9589,9 @@ point game::place_player( const tripoint &dest_loc )
...
+            add_msg( m_info, _("%s to drive."), press_x( ACTION_CONTROL_VEHICLE ) );
```
Comment (AMurkin) [inline src/game.cpp]:
```suggestion
            add_msg( m_info, _( "%s to drive." ), press_x( ACTION_CONTROL_VEHICLE ) );
```

**Comment by Night-Pryanik** [comment]
I'm not sure "Wayfarer" is a suitable name for this trait. It means just "wanderer", without link to vehicle traveling or on-foot traveling, right?

**Comment by Siersha** [comment]
> Can the player still use remote controls to drive inside the driven vehicle? From my understanding of the code remote control is handled elsewhere in the code, so you might want to disable that as well.
This is a good point, I'll include disabling remote controls as well. 

> Does this apply to mounting horses and such?
With the current implementation, it does not apply to mounts, but it should be included.

> I'm not sure "Wayfarer" is a suitable name for this trait. It means just "wanderer", without link to vehicle traveling or on-foot traveling, right?
image
I chose Wayfarer because it means exactly that

**Comment by curstwist** [comment]
Keep in mind you can also have a horse/cart follow you now.  There's also the grocery bot and the rideable mechs.  

I'm not sure this is worth 8 points though, I've done plenty of runs without vehicles, it isn't that big a hindrance depending on your plans.

**Comment by Siersha** [comment]
> Keep in mind you can also have a horse/cart follow you now. There's also the grocery bot and the rideable mechs.

Having something follow you or be dragged around for extra storage to dump someplace isn't what the trait is trying to do. It's just to prevent you from travelling other than on foot.


> I'm not sure this is worth 8 points though

The 8 point rationale stems from a comparison between other traits of similar point values.
Illiterate (6 points): I've done the majority of my playthroughs without reading books without this trait since I tend to play on .25 item spawns. Though that's on me, books are not a hindrance. I'd easily place inability to use vehicles to be worse than inability to read books.
Fragile (8 points): A trait I never play without. Dodge is too good if there's heavy investment in Dex. Barring that, light survivor suit is fairly easy to craft. Playing with these traits in mind, depending on your plans, it's not that much of a hindrance.

>I've done plenty of runs without vehicles, it isn't that big a hindrance depending on your plans.
The same thing can be said for most negative traits, depending on your plans.

**Comment by shadowsapex** [comment]
I feel it would be better to have a much more direct name for the trait, like "can't drive"

**Comment by Siersha** [comment]
> I feel it would be better to have a much more direct name for the trait, like "can't drive"

I'm all for changing the name, but "can't drive" doesn't really include not being able to ride horses or remote control cars (which I still need to implement when I get home after work)

**Comment by Kodiologist** [comment]
Another option is "pedestrian".

**Comment by shadowsapex** [comment]
Why are you including horses

**Comment by Siersha** [comment]
> Another option is "pedestrian".

Pedestrian sounds good too. I wonder how many people would prefer Pedestrian over Wayfarer, might need to do a vote.

> Why are you including horses

Because the point of the trait is to disable all modes of travel other than walking or running. This includes disabling mounts and remote controlled vehicles

**Comment by Night-Pryanik** [comment]
> Because the point of the trait is to disable all modes of travel other than walking or running.

I understand the idea, and it looks very interesting and challenging, but it feels too "gamey", at least for me. I think leaving only "some tragic accident with the car in the past gave you a phobia for them" part of description would be more "fitting". And thus horses, mechs, and any other non-car vehicles could still be a viable mean of transportation. I guess we don't need to fully forbid any non-foot transportation, but rather forbid traveling in cars and especially deathmobiles.

**Comment by Siersha** [comment]
> > Because the point of the trait is to disable all modes of travel other than walking or running.
> 
> I understand the idea, and it looks very interesting and challenging, but it feels too "gamey", at least for me. I think leaving only "some tragic accident with the car in the past gave you a phobia for them" part of description would be more "fitting". And thus horses, mechs, and any other non-car vehicles could still be a viable mean of transportation. I guess we don't need to fully forbid any non-foot transportation, but rather forbid traveling in cars and especially deathmobiles.

This makes sense. I may just change Wayfarer to Pedestrian then and call it a day with the way it is implemented now (plus disabling remote controlled vehicles).

**Pull request closed without merging**